networktest.go

```diff
@@ -235,87 +235,110 @@ func (l *lightningNode) Start(lndError chan error) error {
 	// process errors to the goroutine running the tests.
 	go func() {
 		err := l.cmd.Wait()
+		if err != nil {
+			lndError <- errors.Errorf("%v\n%v\n", err, errb.String())
+		}
 
 		// Signal any onlookers that this process has exited.
 		close(l.processExit)
-
-		if err != nil {
-			lndError <- errors.New(errb.String())
-		}
 	}()
 
-	pid, err := os.Create(filepath.Join(l.cfg.DataDir,
-		fmt.Sprintf("%v.pid", l.nodeID)))
+	// Write process ID to a file.
+	if err := l.writePidFile(); err != nil {
+		l.cmd.Process.Kill()
+		return err
+	}
+
+	// Since Stop uses the LightningClient to stop the node, if we fail to get a
+	// connected client, we have to kill the process.
+	conn, err := l.connectRPC()
 	if err != nil {
+		l.cmd.Process.Kill()
 		return err
 	}
-	l.pidFile = pid.Name()
-	if _, err = fmt.Fprintf(pid, "%v\n", l.cmd.Process.Pid); err != nil {
+	l.LightningClient = lnrpc.NewLightningClient(conn)
+
+	// Obtain the lnid of this node for quick identification purposes.
+	ctxb := context.Background()
+	info, err := l.GetInfo(ctxb, &lnrpc.GetInfoRequest{})
+	if err != nil {
+		return err
```

Comment:
Do we need to issue shutdown commands if we fail here or the next error?

Reply:
Yes, good point.

Reply:
Actually, I took another look at it and I think it's better to just ensure that `Teardown` is called if `Start` is ever called, which performs more cleanup, so that Start does not need to worry about it once the `lightningClient` has been set.